experimental/basisLieHighestWeight/MB.jl

```diff
@@ -0,0 +1,364 @@
+        sub_weights = compute_sub_weights(hw)
+        l = length(sub_weights)
+        # go through partitions lambda_1 + lambda_2 = hw until we have monomials enough monomials or used all partitions
+        while length(set_mon) < gapDim && i < l 
+            i += 1
+            lambda_1 = sub_weights[i]
+            lambda_2 = hw .- lambda_1
+            mon_lambda_1 = compute_monomials(t, n, L, lambda_1, ops, wts, wts_eps, monomial_order, calc_hw, cache_size, parallel, no_minkowski)
+            mon_lambda_2 = compute_monomials(t, n, L, lambda_2, ops, wts, wts_eps, monomial_order, calc_hw, cache_size, parallel, no_minkowski)
+            mon_sum = Set([p .+ q for p in mon_lambda_1 for q in mon_lambda_2]) # Minkowski-sum: M_{lambda_1} + M_{lambda_2} subseteq M_{hw}
+            union!(set_mon, mon_sum)
+        end
+        # check if we found enough monomials
+        if length(set_mon) < gapDim
+            push!(no_minkowski, hw)
+            set_mon = add_by_hand(t, n, L, hw, ops, wts, wts_eps, monomial_order, gapDim, set_mon, cache_size, parallel)
+        end
+        push!(calc_hw, hw => set_mon)
+        return set_mon
+    end
+end
+
+function is_fundamental(hw)
+    """
+    returns true if hw is fundamental weight, i.e. hw = [0, ..., 1, ..., 0] (or hw = [0, ..., 0])
+    """
+    one = false
+    for i in hw
+        if i > 0
+            if one || i > 1
+                return false
+            else 
+                one = true
+            end
+        end
+    end
+    return true
+end
+
+function order_sub_weights(hw, sub_weights)
+    """
+    sort weights incrasing by their 2-norm
+    """
+    sort!(sub_weights, by=x->sum((x).^2))
+end
+
+function compute_sub_weights(hw)
+    """
+    returns list of weights w != 0 with 0 <= w <= hw elementwise
+    """
+    sub_weights = []
+    foreach(Iterators.product((0:x for x in hw)...)) do i
+        push!(sub_weights, [i...])
+    end
+    popfirst!(sub_weights) # [0, ..., 0]
+    pop!(sub_weights) # hw
+    order_sub_weights(hw, sub_weights)
+    return sub_weights
+end
+
+function add_known_monomials!(weightspace, set_mon_in_weightspace, m, wts, mats, calc_monomials, space, e, cache_size)
+    """
+    By using the Minkowski-sum, we know that all monomials in set_mon_in_weightspace are in our basis. Since we want to extend the weightspacse with
+    missing monomials, we go need to calculate and add the vector of each monomial to our basis.
+    """
+    weight = weightspace[1]
+    for mon in set_mon_in_weightspace[weight]
+        #vec, wt = calc_new_mon!(mon, m, wts, mats, calc_monomials, space, e, cache_size)
+        d = sz(mats[1])
+        v0 = spzeros(Int, d); v0[1] = 1 # starting vector v
+        vec = calc_vec(v0, mon, mats)
+        wt = calc_wt(mon, wts)  
+        #println("vec:" , vec)
+        #println("wt: ", wt)
+        if !haskey(space, wt)
+            space[wt] = nullSpace()
+        end
+        addAndReduce!(space[wt], vec)
+    end
+    GC.gc()
```

Comment:
Why?

Reply:
I don't know why, but the garbage collector of Julia sometimes doesn't delete the dereferenced objects of the function add_known_monomial and add_new_monomials. Therefore, I had memory problems without calling it manually. I did not yet find a minimal working example and can't figure out the cause. After completing the other major changes, I will test it again and try to find another solution.